packages/rum/src/boot/recorderApi.ts

```diff
@@ -139,9 +142,11 @@ export function makeRecorderApi(
         return cachedDeflateEncoder
       }
 
-      startStrategy = () => {
+      startStrategy = (options?: StartStrategyOptions) => {
         const session = sessionManager.findTrackedSession()
-        if (!session || !session.sessionReplayAllowed) {
+        // If session is undefined (untracked), recording should not start
+        // If session is not sampled for replay and not being forced, recording should not start
+        if (!session || (!session.sessionReplayAllowed && !options?.forceStart)) {
```

Comment:
I separated the state now into two, one that reflects if the session was originally sampled in for replay, and the second one to check if replay is available for the session

Reply:
> Also you raise a good point. We should agree on what is intended to be conveyed in view.session.sampled_for_replay and if we want to make the distinction with force replay.

In fact, a forced session should still remain with `sampled_for_replay: false` but has `has_replay: true`. With the change I added, for a forced replay, it'll have `sessionReplayAllowed: true` but `sampled_for_replay: false`

Reply:
I wondering if it can make investigations a bit harder. I guess if we want to distinguish `sampled_for_replay` and force replay, we should also add another field  like `replay_forced` in the event to reflect it.  cc @bcaudan, @thomas-lebeau

Reply:
As discussed (according to the approved RFC), this is the expected behaviour. Resolving the comment accordingly